bftengine/tests/testViewChange/testViewChange.cpp

```diff
@@ -52,28 +52,36 @@ class myViewsManager : public ViewsManager {
 class DummyReplica : public InternalReplicaApi {
  public:
   DummyReplica(const bftEngine::impl::ReplicasInfo& replicasInfo, const ReplicaConfig& replicaConfig)
-      : replicasInfo_(replicasInfo),
+      : operationStatus_(PrepareCombinedSigOperationStatus::OPERATION_PENDING),
+        replicasInfo_(replicasInfo),
         replicaConfig_(replicaConfig),
         msgHandlersPtr_(new MsgHandlersRegistrator()),
         incomingMsgsStorage_(new IncomingMsgsStorageImp(msgHandlersPtr_, timersResolution, replicaConfig_.replicaId)) {
-    combine_operation_future = combine_operation_promise.get_future();
     SeqNumInfo::init(seqNumInfo_, static_cast<void*>(this));
+    dynamic_cast<IncomingMsgsStorageImp*>(incomingMsgsStorage_)->start();
+  }
+  ~DummyReplica() {
+    dynamic_cast<IncomingMsgsStorageImp*>(incomingMsgsStorage_)->stop();
+    delete incomingMsgsStorage_;
   }
-  ~DummyReplica() { delete incomingMsgsStorage_; }
 
   void onPrepareCombinedSigFailed(SeqNum seqNumber, ViewNum view, const set<uint16_t>& replicasWithBadSigs) {
-    combine_operation_promise.set_value(PrepareCombinedSigOperationStatus::OPERATION_FAILED);
+    std::unique_lock<std::mutex> lock(m_);
+    operationStatus_ = PrepareCombinedSigOperationStatus::OPERATION_FAILED;
+    cv_.notify_one();
```

Comment:
what's the purpose of `cv_` ?

Reply:
It replaces the future/promise objects used in previous version. Here we have multiple calls to sinc, because when we add msg-s to the seqNumInfo object it checks if enough signature shares have arrived to combine them. This combination happens on the thread pool of the replica, and after it is completed the thread that has made the combination calls either onPrepareCombinedSigSucceeded(), or onPrepareCombinedSigFailed(), so basically after the main thread has added enough signature shares it has to wait for the combining thread to actually combine them and generate PrepareFullMsg.

Reply:
Then you should simply include this explanation in the PR's description, so that reviewers can find their way around the changes...

Reply:
Using mock for threshold signatures generation/verification makes the mock-ed Replica no longer necessary.